### PR TITLE
PullRequst for Issue9: add message type MiddleTime plus Count Before and Afte

### PR DESCRIPTION
**Pull request by XuetaoLiu-S**
Add the feature for the MiddleTimePlusCountBeforeAndAfter

https://github.com/acquaman/AcquamanDataServer/issues/9

### CODE REVIEW

**Comment by iainworkman** [comment]
Again, this looks good to me.
Time Estimate: 30 minutes

File: source/AMDSBufferGroup.cpp [hunk 1]
```diff
@@ -114,6 +119,20 @@ void AMDSBufferGroup::populateData(AMDSClientStartTimeToEndTimeDataRequest* clie
 	}
 }
 
+void AMDSBufferGroup::populateData(AMDSClientMiddleTimePlusCountBeforeAndAfterDataRequest* clientDataRequest)
+{
+	QDateTime middleTime = clientDataRequest->middleTime();
+	int countBefore = clientDataRequest->countBefore();
+	int countAfter = clientDataRequest->countAfter();
+
+	int middleIndex = lowerBound(middleTime);
```
Comment (dretrex) [inline source/AMDSBufferGroup.cpp:132]: I don't really like the name `lowerBound()`.  It doesn't convey a lot of information.  For instance, what is an `int` that is returned a lower bound of a QDateTime?

**Comment by dretrex** [comment]
Overall, code looks good.  However, what is this request?  I can't figure it out based on the name.
Time estimate: 5 minutes

**Comment by XuetaoLiu-S** [comment]
@dretrex this request is for the given [count] data before the given time and the the given [count] data after the given time.

Regarding the `lowerBount()`, I will rename it in a later branch, otherwise I will have too many merge conflict.

**Pull request merged**